integration-tests/tests/src/utils/expect-sync-error.ts

```diff
@@ -0,0 +1,85 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import { openRealm } from "./open-realm";
+
+/**
+ * Open a new Realm and perform an action, expecting a sync error to occur. Will
+ * never resolve and therefore timeout if a sync error does not occur.
+ *
+ * @param config The Realm config to use
+ * @param user The Realm user to use
+ * @param action Callback receiving a Realm instance and containing the action(s) to
+ * take which should trigger a sync error
+ * @param expectation Callback receiving the sync error, in order to make assertions about it
+ * @returns Promise which resolves if the sync error occurs
+ */
+export const expectSyncError = async (
```

Comment:
Any particular reason these are arrow functions and not simply `function`s?

Reply:
Purely force of habit, I suspect! I think it would make sense to be consistent within one file though so I'll update